### ISSUE TITLE
New user creation -- invalid rotki api key shows no error

### ISSUE THREAD

**Issue by LefterisJP** [rotki/rotki#2856]
## Problem Definition

If at new user creation,the user enters an invalid api premium key the frontend does not show an error message. It just throws the user back to the login screen.

Looking at the logs an error is returned by the backend but the frontend does not seem to do anything.

Demo:

https://user-images.githubusercontent.com/1658405/117322086-fb9e2800-ae8d-11eb-8a60-4c04f7766275.mp4

Logs:

```
[06/05/2021 17:01:22 CEST] INFO rotkehlchen.rotkehlchen: Unlocking user user=test_foo1, create_new=True, sync_approval=yes, initial_settings=ModifiableDBSettings(premium_should_sync=None, include_crypto2crypto=None, anonymized_logs=None, ui_floating_precision=None, taxfree_after_period=None, balance_save_frequency=None, include_gas_costs=None, eth_rpc_endpoint=None, ksm_rpc_endpoint=None, main_currency=None, date_display_format=None, submit_usage_analytics=False, kraken_account_type=None, active_modules=None, frontend_settings=None, account_for_assets_movements=None, btc_derivation_gap_limit=None, calculate_past_cost_basis=None, display_date_in_localtime=None, current_price_oracles=None, historical_price_oracles=None, taxable_ledger_actions=None)
[06/05/2021 17:01:22 CEST] ERROR rotkehlchen.premium.sync: Given API key is invalid
[06/05/2021 17:01:22 CEST] DEBUG rotkehlchen.api.rest: Request successful response={"result": null, "message": "Could not verify keys for the new account. Rotkehlchen API key was rejected by server"}, status_code=409
[06/05/2021 17:01:22 CEST] INFO rotkehlchen.api.server.pywsgi: 127.0.0.1 - - "PUT /api/1/users HTTP/1.1" 409 311 0.627478
[06/05/2021 17:01:30 CEST] DEBUG rotkehlchen.server: Shutdown initiated
```

This happens both with an invalid api key format (as in the demo - status code 400) and a key rejected by server (logs - status codes 409)

## Task

Fix it, and show an error to the user saying that the given premium api keys are invalid or have a bad format or whatever the error from the backend says.

**Comment by kelsos** [comment]
@LefterisJP the key fails but the account still gets created right? so the user should be logged in and see the error.

**Comment by CruiseMan** [comment]
Create user for first time gives this error


https://user-images.githubusercontent.com/15807643/117354767-42494d80-aea1-11eb-8ffb-212b3594a1f6.mp4

**Comment by LefterisJP** [comment]
Fixed by #2862
